deals-add.js

```diff
@@ -0,0 +1,280 @@
+		 */
+		resetFeed: function () {
+			Config.offset = 0;
+			Config.hasMore = true;
+			Config.dealsContainer.firstChild.style.visibility='hidden';
+			const j = 0;
+			for (j = document.getElementsByClassName("deal-card").length; j > 1; j--) {
+				Config.dealsContainer.removeChild(Config.dealsContainer.lastChild);
+			}
+		},
+		/**
+		 * Populate deal cards with information fetched from RateS endpoint
+		 *
+		 * @param {Object}    response				contains data array with product information to populate deal cards with.
+		 * @param {Number}		cardNumber			the card to populate the information 			
+		 * @param {Number}		dataEntry				the entry in the data array to populate the cards with.
+		 */
+		populateDeals: function (response, cardNumber, dataEntry) {
+			// set product URL
+			document.getElementsByClassName("deal-link")[cardNumber].href = response.data[dataEntry].listing.merchantURL;
+					
+			// set product image
+			document.getElementsByClassName("deal-img")[cardNumber].src = response.data[dataEntry].images[0];
+			
+			// set merchant of product and change it to Title case
+			document.getElementsByClassName("merchant")[cardNumber].innerHTML = "at " + toTitleCase(response.data[dataEntry].listing.merchant);
+
+			// set deal title 
+			const dealTitle = document.getElementsByClassName("deal-item-title")[cardNumber];
+			dealTitle.innerHTML = response.data[dataEntry].name;
+			// if deal title is too long, clamp it to show ellipsis
+			clamp(dealTitle); 
+			
+			/*
+			* commented out for now, may be needed for activity feed in the future
+			//set how long ago deal was posted
+			document.getElementsByClassName("deal-posted-date")[cardNumber].innerHTML = getTimeAgo(response.data[dataEntry].lastCreated);
+			*/
+
+			// set current price with correct currency
+			document.getElementsByClassName("current-price")[cardNumber].innerHTML = getCurrency(response.data[dataEntry]) + round(response.data[dataEntry].listing.currentPrice, 2);
+			
+			// set savings with correct currency and decimal format
+			if (response.data[startOfData].listing.previousPrice !== "") { 
+				// if there are savings, calculate and set
+				document.getElementsByClassName("save-container")[cardNumber].lastChild.style.visibility='visible';
+				document.getElementsByClassName("prices-container")[cardNumber].lastChild.style.visibility='visible';
+				document.getElementsByClassName("amount-saved")[cardNumber].innerHTML = getCurrency(response.data[dataEntry]) + calculateSavings(response.data[dataEntry]);
+			}
+			else { 
+				// if no savings, hide savings related elements
+				document.getElementsByClassName("save-container")[cardNumber].lastChild.style.visibility='hidden';
+				document.getElementsByClassName("prices-container")[cardNumber].lastChild.style.visibility='hidden';
+			}
+
+			// to account for first card already on the page
+			if (cardNumber < response.data.length - 1) {
+				createNewCard();
```

Comment:
You can't call the function like that anymore. Will need to do it like `dealsNamespace.createNewCard()`.

Reply:
This will be undefined